### ISSUE TITLE
Option source-map-map-inline produces an empty file

### ISSUE THREAD

**Issue by nicoschoenmaker** [less/less.js#2987]
Using the --source-map-map-inline option produces an empty file.

Steps to reproduce:
- Go to an empty directory
- Make sure we have node + less
```bash
wget https://nodejs.org/dist/v7.1.0/node-v7.1.0-linux-x64.tar.xz
tar -xf node-v7.1.0-linux-x64.tar.xz
./node-v7.1.0-linux-x64/bin/node ./node-v7.1.0-linux-x64/bin/npm install less
echo ".red { color: blue; }" > foo.less
```
- Process the just created `foo.less` file:
```
./node-v7.1.0-linux-x64/bin/node node_modules/.bin/lessc foo.less --source-map-map-inline
```

Expected
```
.red {
  color: blue;
}
/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImZvby5sZXNzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUFBO0VBQU8sV0FBQSIsImZpbGUiOiJmb28uY3NzIn0= */
```
Actual result: only a newline

Sidenotes:
- Works fine without the --source-map-map-inline parameter
- This was already reported in #2961, but closed since the author found a workaround. Which is to specify the destination:
```
./node-v7.1.0-linux-x64/bin/node node_modules/.bin/lessc foo.less --source-map-map-inline foo.css
```

Why is it a bug? In the docs the destination (foo.css) is specified as an optional argument. For my use case it would be a lot faster if I did not have to create a temporary file.

**Comment by rickprent** [comment]
+1

**Comment by megahallon** [comment]
I believe you can also close #2897 as it is described now, however you will also get this issue (TypeError on path) when doing: `lessc --source-map=foo foo.less`

**Comment by nicoschoenmaker** [comment]
This issue was fixed with #2988
